src/components/breadcrumbs/_breadcrumb_content.styles.ts

```diff
@@ -17,12 +17,23 @@ import {
   logicalBorderRadiusCSS,
   mathWithUnits,
 } from '../../global_styling';
+import { euiButtonColor } from '../../themes/amsterdam/global_styling/mixins/button';
 
 /**
  * Styles cast to inner <a>, <button>, <span> elements
  */
 export const euiBreadcrumbContentStyles = (euiThemeContext: UseEuiTheme) => {
-  const { euiTheme } = euiThemeContext;
+  const { euiTheme, colorMode } = euiThemeContext;
+  const breadcrumbButtonColor = euiButtonColor(euiThemeContext, 'primary');
+  const breadcrumbTextColors = {
```

Comment:
Some very nitty comments (all optional!):

```suggestion
  const { euiTheme, colorMode } = euiThemeContext;

  // Reuse button colors for `type="application`" clickable breadcrumbs
  const applicationButtonColors = euiButtonColor(euiThemeContext, 'primary');

  // Create our own custom darker gray colors for non-clickable application breadcrumbs
  // The numbers/ratios are fairly specific here to pass WCAG AA contrast minimums
  const applicationTextColors = {
```

I suggested the above changes for a few reasons:

- this whole file is about breadcrumbs so I think that naming is kind of a given - renaming it to `application` instead is I think more helpful to hint where exactly these color vars will be used
- The comments err on the verbose side, but I think help answer questions for devs in another 6 years, who may be like "why did they choose 0.85 here specifically?" And while yes, they could dig through git blame to find this originating PR, file history may have changed by then (entirely new file, no longer on Emotion, etc) to make this harder to track down, whereas a comment is a bit easier to get an immediate "why" from.

Reply:
Thanks, yeah that makes sense. It's better to add more context - I'll update!